.github/actions/VerifyAppChanges/TestPreprocessorSymbols.ps1

```diff
@@ -0,0 +1,47 @@
+# Current path is .github/actions/VerifyAppChanges
+
+Import-Module "$PSScriptRoot\..\..\..\build\scripts\EnlistmentHelperFunctions.psm1" -DisableNameChecking
+Import-Module "$PSScriptRoot\..\..\..\build\scripts\GuardingV2ExtensionsHelper.psm1" -DisableNameChecking
+Import-Module "$PSScriptRoot\..\..\..\build\scripts\TestPreprocessorSymbols.psm1" -Force
+
+# Get the major build version from the main branch
+$mainVersion = Get-MaxAllowedObsoleteVersion
+
+# Get the major build version from the current branch
+$currentVersion = (Get-ConfigValue -Key "repoVersion" -ConfigType AL-Go) -split '\.' | Select-Object -First 1
+
+# CLEANSCHEMA is on a 5y cycle starting from 26
+if ($CurrentVersion -le 26) {
+    $schemaLowerBound = 15
+} else {
+    $schemaLowerBound = ([math]::Floor(($CurrentVersion - 2) / 5) * 5) - 2
+}
+
+# Define the preprocessor symbols to check for
+$symbolConfigs = @(
+    @{stem = "CLEAN"; lowerBound = ($CurrentVersion - 4); upperBound = $mainVersion},
```

Comment:
Why not use this list of CLEAN tags? 
https://github.com/microsoft/BCApps/blob/1a67cfd41f42a3a58efce7a3fa1c8c92993e5d08/.github/AL-Go-Settings.json#L12C1-L23C5

If we calculate it like this based on the current version, we might be blocked from doing version updates

Reply:
Yes, that is kind of the point. There should be no old tags left behind because no one will look at them, and this ensure that. Take the current state as example: We have main = 26 so that is the upperbound, and 22 as the lowerbound. We just cleaned up all v23 tags... When you move to 27, 23 will become the lowerbound... If there is any tag 22 or lower left then that would fail... that also means we had 2 releases to clean it up... At that point things should fail so they get cleaned up.

Reply:
Personally, I'm not a fan of blocking app version updates because we forgot to remove some old code. Queues are typically blocked when we update the app version in main so we want to minimize the amount of time it takes before queues can be opened again. 

Couldn't you achieve the same thing by: 
* Updating the allowed CLEAN versions when you clean up code. 
* Running this check against the list of allowed CLEAN tags

That way you make sure we clean up old code without blocking app version updates.

Reply:
You will run into the exact same issue because now you rely on a human to update the list... and the human will forget. It doesn't solve the problem.